internal/store/pod.go

```diff
@@ -237,6 +237,31 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_pod_status_unscheduled",
```

Comment:
this is a condition happening during scheduling right? if so wouldn't this be "unschedulable"?

Reply:
yeah, this is a condition happening during scheduling. As described in https://github.com/kubernetes/kube-state-metrics/issues/829,the condition type is `PodScheduled`,but the status is `False`, because of insufficient resource or specific node selector or node affinity strategy.

Reply:
Yes I'm just saying `unscheduled` has a different meaning than `unschedulable`. The first is something eviction like, the latter describes this situation, so we should change the metric name to `kube_pod_status_unschedulable`.

Reply:
I do wonder if we need a separate metric at all though, as we already have a metric that reflects a pod being in pending phase.

Reply:
`kube_pod_status_unschedulable` will be a better metric name,is there a way to describe the state that pod can't be scheduled because of lack of resources without adding this metric?If there is a way, the `kube_pod_status_unschedulable` shouldn't be added.

Reply:
kube_pod_status_phase{phase=“Pending”} should capture the same. If this metric is the active one for 1min then the Pod can’t be scheduled.

Reply:
if the image is big, the pulling state is also in `kube_pod_status_phase{phase=“Pending”}`, the `Pending` state includes both image pulling and unschedulable.My origin intention is to distinguish the two state.

Reply:
I see. In that case this should be ok to add.